hangupsbot/plugins/tldr.py

```diff
@@ -35,9 +35,16 @@ def tldrecho(bot, event, *args):
     else:
         # No path was found. Is this your first setup?
         new_tldr = 0
+    
+    if tldr_echo_options[new_tldr] is not "GLOBAL":
+        # Update the tldr_echo setting
+        bot.memory.set_by_path(['conversations', event.conv_id, 'tldr_echo'], new_tldr)
+    else:
+        # If setting is global then clear the conversation memory entry
+        conv_settings = bot.memory.get_by_path(['conversations', event.conv_id])
+        del conv_settings['tldr_echo'] # remove setting
+        bot.memory.set_by_path(['conversations', event.conv_id], conv_settings)
```

Comment:
That's fine - 0 is the next iteration anyway, and removing it means there's no memory entry so it defaults back to there global setting in config, which is what was being ignored.

Either you put in a special case in the code setting the value to delete it from memory when new_tldr == 2, or you have even messier code handling the fact that there are two situations in which the global setting needs to be used.

For ease of coding, and common sense in terms of interpreting the behaviour, littering memory.json with entries which just tell the code to do what is set in config seems less clean than just deleting the memory entry and relying on the default.

memory.json should (I thought) be containing changes from the default config setting in this instance, not reiterating them.